app/ios/ioshandler.cpp

```diff
@@ -0,0 +1,28 @@
+#include <QCoreApplication>
+#include <QPointer>
+#include <QtCore>
+#include <QDebug>
+
+#include "ioshandler.h"
+
+static IOSHandler *mInstance;
+
+IOSHandler::IOSHandler( QObject *parent ) : QObject( parent )
+{
+
+}
+
+IOSHandler *IOSHandler::instance()
```

Comment:
in general, singleton is anti-pattern.. hard to test, hard to release resource on end (leaks), possible initialization fiasko etc. 

if technically possible I would avoid it (for example it would be created in main input app function after initialization of qgsapplication?)

Reply:
The IOSHandler has been deleted and functionality moved to IOSImagePicker which has to be passed to view delegate for callbacks.